### PR TITLE
fix Views field custom date field output for year only

### PR DESCRIPTION
**Pull request by jackrabbithanna**
Port and tweak of contributed PR https://github.com/eileenmcnaughton/civicrm_entity/pull/435

Before
----------------------------------------
Year only custom date fields for some time zones displays wrong value. 

After
----------------------------------------
Custom fields with year only format always show correctly.

### CODE REVIEW

**Comment by jackrabbithanna** [comment]
Difference between this and contributed PR for 3.x branch is to always use date_default_timezone_get() and not the system-wide timezone configuration value

File: src/Plugin/views/field/CustomEntityField.php [hunk 1]
```diff
@@ -335,4 +334,40 @@ protected function getDelta($id) {
...
+   *   The field definition.
+   * @param string $date_value
+   *   The date value.
+   *
+   * @return string
+   *   The converted value.
+   */
+  public function convertToUtc(FieldDefinitionInterface $definition, $date_value) {
+    $datetime_format = $definition->getSetting('datetime_type') === DateTimeItem::DATETIME_TYPE_DATE ? DateTimeItemInterface::DATE_STORAGE_FORMAT : DateTimeItemInterface::DATETIME_STORAGE_FORMAT;
+    $default_timezone = date_default_timezone_get();
```
Comment (jackrabbithanna) [inline src/Plugin/views/field/CustomEntityField.php:350]: This line is the difference from PR #435 for the 3.x branch

Comment (jackrabbithanna) [reply src/Plugin/views/field/CustomEntityField.php:350]: If using the global configuration then if the per-user timezone setting is allowed, and if the user chooses a different timezone, then could have a discrepancy. Drupal manages this and gives the "right" timezone returned by  date_default_timezone_get();

**Comment by jitendrapurohit** [comment]
We've applied this PR on production and can confirm this is working for year-only custom fields.

**Review by jitendrapurohit — APPROVED** [review]

**Pull request merged**